tests/unit/s2n_post_handshake_send_test.c

```diff
@@ -0,0 +1,235 @@
+        EXPECT_OK(s2n_connection_set_secrets(server_conn));
+
+        DEFER_CLEANUP(struct s2n_connection *client_conn = s2n_connection_new(S2N_CLIENT), s2n_connection_ptr_free);
+        EXPECT_SUCCESS(s2n_connection_set_config(client_conn, config));
+        EXPECT_SUCCESS(s2n_connection_set_all_protocol_versions(client_conn, S2N_TLS13));
+        EXPECT_OK(s2n_connection_set_secrets(client_conn));
+
+        DEFER_CLEANUP(struct s2n_test_io_stuffer_pair io_pair = { 0 }, s2n_io_stuffer_pair_free);
+        EXPECT_OK(s2n_io_stuffer_pair_init(&io_pair));
+        EXPECT_OK(s2n_connections_set_io_stuffer_pair(client_conn, server_conn, &io_pair));
+
+        /* Calculate the size of the NewSessionTicket message */
+        uint32_t nst_size = 0;
+        EXPECT_OK(s2n_get_nst_message_size(server_conn, &nst_size));
+
+        /* Test: the messages are fragmented into the number of records expected */
+        uint64_t write_seq_num = 0;
+        for (size_t frag_i = 0; frag_i < s2n_array_len(fragment_sizes); frag_i++) {
+            const uint32_t fragment_size = fragment_sizes[frag_i];
+
+            server_conn->max_outgoing_fragment_length = fragment_size;
+            server_conn->tickets_sent = 0;
+            server_conn->tickets_to_send = tickets_to_send;
+
+            EXPECT_SUCCESS(s2n_send(server_conn, send_data, sizeof(send_data), &blocked));
+            EXPECT_EQUAL(server_conn->tickets_sent, tickets_to_send);
+            EXPECT_TRUE(s2n_stuffer_is_freed(&server_conn->handshake.io));
+
+            uint64_t actual_record_count = 0;
+            uint64_t expected_record_count = 0;
+            uint8_t *seq_num = server_conn->server->server_sequence_number;
+            EXPECT_OK(s2n_get_actual_record_count(seq_num, &write_seq_num, &actual_record_count));
+            EXPECT_OK(s2n_get_expected_record_count(nst_size, fragment_size, tickets_to_send, &expected_record_count));
+            EXPECT_EQUAL(actual_record_count, expected_record_count);
+        }
+
+        /* Test: the full messages can be parsed by the client */
+        uint64_t read_seq_num = 0;
+        for (size_t frag_i = 0; frag_i < s2n_array_len(fragment_sizes); frag_i++) {
+            const uint32_t fragment_size = fragment_sizes[frag_i];
+
+            uint8_t recv_data[sizeof(send_data)] = { 0 };
+            EXPECT_SUCCESS(s2n_recv(client_conn, recv_data, sizeof(recv_data), &blocked));
+            EXPECT_BYTEARRAY_EQUAL(send_data, recv_data, sizeof(recv_data));
+
+            uint64_t actual_record_count = 0;
+            uint64_t expected_record_count = 0;
+            uint8_t *seq_num = client_conn->server->server_sequence_number;
+            EXPECT_OK(s2n_get_actual_record_count(seq_num, &read_seq_num, &actual_record_count));
+            EXPECT_OK(s2n_get_expected_record_count(nst_size, fragment_size, tickets_to_send, &expected_record_count));
+            EXPECT_EQUAL(actual_record_count, expected_record_count);
+        }
+    }
```

Comment:
For clang-format and consistency add `;` at the end of blocks. Can't grep for these, but we need a simple-check.